### ISSUE TITLE
Incorrect findings for HorizontalPodAutoscaler

### ISSUE THREAD

**Issue by sybo-matthew** [k8sgpt-ai/k8sgpt#216]
Checklist:

* [x] I've searched for similar issues and couldn't find anything matching
* [x] I've included steps to reproduce the bug.
* [x] I've included the version of Kubernetes and k8sgpt.

### Subject of the issue

k8sgpt returns incorrect information reporting missing information about HorizontalPodAutoscaler. Specifically: `- Error: HorizontalPodAutoscaler uses Deployment/teams as ScaleTargetRef which does not exist.` 

### Your environment

* Version of Kubernetes
Client Version: v1.24.12-dispatcher
Kustomize Version: v4.5.4
Server Version: v1.24.9-gke.3200
* MacOS 13.2.1
* Version of k8sgpt
k8sgpt version 0.1.8

### Steps to reproduce

* Run `k8sgpt analyze --explain`
* Receive output of: `- Error: HorizontalPodAutoscaler uses Deployment/teams as ScaleTargetRef which does not exist.`
* Confirm deployment does exist in namespace. 
``` kubectl get deployment teams -n other-namespace
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
teams   2/2     2            2           144d
```
* Confirm autoscaling is working
`Name:                                                  teams
Namespace:                                          other-namespace
Annotations:                                           <none>
CreationTimestamp:                              Fri, 11 Nov 2022 11:07:40 +0100
Reference:                                             Deployment/teams
Metrics:                                               ( current / target )
  resource cpu on pods  (as a percentage of request):  1% (2m) / 80%
Min replicas:                                          2
Max replicas:                                          150
Deployment pods:                                       2 current / 2 desired
Conditions:
  Type            Status  Reason            Message
  ----            ------  ------            -------
  AbleToScale     True    ReadyForNewScale  recommended size matches current size
  ScalingActive   True    ValidMetricFound  the HPA was able to successfully calculate a replica count from cpu resource utilization (percentage of request)
  ScalingLimited  True    TooFewReplicas    the desired replica count is less than the minimum replica count
Events:           <none>`

### Expected behaviour
k8sgpt should not be trying to check the HPA at all. 

`❯ k8sgpt filters list
Active:
> Pod
> ReplicaSet
> PersistentVolumeClaim
> Service
> Ingress
Unused:
> HorizontalPodAutoScaler
> PodDisruptionBudget`

### Actual behaviour
It seems to be incorrectly checking for HPA deployment targets in the default namespace.

**Comment by matthisholleville** [comment]
Hello,

Thank you for your issue. Are your HPA and Deployment located in the same namespace?

**Comment by matthisholleville** [comment]
I just tried it and I couldn't reproduce it. Are your deployment and HPA in the same namespace?

**Comment by sybo-matthew** [comment]
Yeah if you see the output from the HPA it’s in the same namespace as the service.

**Comment by matthisholleville** [comment]
Can you please send me the ApiVersions that you are using for HPA and Deployment?

**Comment by larssb** [comment]
I'm also seeing this:

HPA API version: autoscalign/v2
Deployment: apps/v1

I'm talking about this issue in [#229](https://github.com/k8sgpt-ai/k8sgpt/issues/229)

**Comment by matthisholleville** [comment]
Thank you for reporting the issue. I have just pushed a fix that resolves the problem.
